### PR TITLE
Allow to specify the environment variables for the process

### PR DESCRIPTION
**Pull request by xavierhainaux**
This allows to pass the variable `HOME` or  `APPLOCALDATA` to control where the analysis-server should put the temporary files (the `.dartServer` folder).

If you know a better way, I'm all for it :-)

### CODE REVIEW

**Comment by devoncarew** [comment]
> This allows to pass the variable HOME or APPLOCALDATA to control where the analysis-server should put the temporary files (the .dartServer folder).

Is this something you need to control? Generally this should be a black box to clients of the analysis server. I'm not sure that controlling it indirectly by setting `HOME` or `APPLOCALDATA` is a great idea :(

File: tool/generate_analysis.dart [hunk 1]
```diff
@@ -53,7 +53,8 @@ main(List<String> args) {
   DartGenerator generator = new DartGenerator();
   api.generate(generator);
   outputFile.writeAsStringSync(generator.toString());
-  Process.runSync('dartfmt', ['-w', outputFile.path]);
+  Process.runSync(
+      'dartfmt${Platform.isWindows ? ".bat" : ""}', ['-w', outputFile.path]);
```
Comment (devoncarew) [inline tool/generate_analysis.dart]: this change lgtm :)

**Comment by xavierhainaux** [comment]
We run serveral analysis-servers in parallel on the same machine (sometime with different version of the dart-sdk).

We observed that the performance of analysis was more predictable if each instance had their own cache folder. I don't have any measurement for this, this is just a small feeling.

Maybe we could ask the analyzer to have an official flag to control this location: https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/lib/file_system/physical_file_system.dart#L104

**Comment by devoncarew** [comment]
Passing in the env var sgtm.

For your use case, I would ask for a specific cli flag on the analysis server, just for reliability (the env var technique may stop working). It's a pretty rare use case, so the analysis server team may say no. You'd want to open an issue here: https://github.com/dart-lang/sdk.

**Comment by devoncarew** [comment]
And thanks for the PR!

**Pull request merged**